### ISSUE TITLE
Investigate regression in memory usage in v7

### ISSUE THREAD

**Issue by CharlieC3** [hirosystems/stacks-blockchain-api#1564]
We're seeing more OOMs in v7 than we did in v6. We may need to profile the memory usage again to identity key areas for performance improvements and address them.

**Comment by rafaelcr** [comment]
We should perform tests in the new clusters by reducing pod memory in staging and sending prod traffic into it so we can get a memory dump once it crashes

**Comment by zone117x** [comment]
Is this still an issue? Did we fix it by switching to the slim docker image and upgrading nodejs version?

**Comment by CharlieC3** [comment]
@zone117x Yeah this is still an issue unfortunately. Changing the LB algo to least_request helped cover the symptoms, but OOMs are still occurring frequently, just less so than before.

**Comment by rafaelcr** [comment]
Repurposed this issue to investigate the high memory usage in API readers, possibly caused by users reading very large transactions that are currently decoded as a single string from postgres (`bytea` -> `string` decoding uses a ton of memory).
